rotkehlchen/tests/api/test_nfts.py

```diff
@@ -948,3 +949,84 @@ def mock_uniswap_v3_balances(*_args, **_kwargs) -> dict[str, Any]:
+                        floor_price=FVal(0.00098),
+                    ),
+                ),
+            ],
+            TEST_ACC4: [
+                NFT(
+                    token_identifier='_nft_0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85_26612040215479394739615825115912800930061094786769410446114278812336794170041',  # noqa: E501
+                    background_color=None,
+                    image_url='https://openseauserdata.com/files/3f7c0c7d1ba51e61fe05ef53875f9f7e.svg',  # noqa: E501
+                    name='yabir.eth',
+                    external_link='https://app.ens.domains/name/yabir.eth',
+                    permalink='https://opensea.io/assets/ethereum/0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85/26612040215479394739615825115912800930061094786769410446114278812336794170041',  # noqa: E501
+                    price_eth=FVal(0.00098),
+                    price_usd=FVal(1.2379458),
+                    collection=Collection(
+                        name='ENS: Ethereum Name Service',
+                        banner_image=None,
+                        description='Ethereum Name Service (ENS) domains are secure domain names for the decentralized world. ENS domains provide a way for users to map human readable names to blockchain and non-blockchain resources, like Ethereum addresses, IPFS hashes, or website URLs. ENS domains can be bought and sold on secondary markets.',  # noqa: E501
+                        large_image='https://i.seadn.io/gae/BBj09xD7R4bBtg1lgnAAS9_TfoYXKwMtudlk-0fVljlURaK7BWcARCpkM-1LGNGTAcsGO6V1TgrtmQFvCo8uVYW_QEfASK-9j6Nr?w=500&auto=format',  # noqa: E501
+                        floor_price=FVal(0.00098),
+                    ),
+                ),
+            ],
+        }
+        return data, 2
+
+    get_all_nft_data_patch = patch('rotkehlchen.chain.ethereum.modules.nft.nfts.Nfts._get_all_nft_data', side_effect=mock_get_all_nft_data)  # noqa: E501
+
+    with get_all_nft_data_patch:
+        response = requests.get(  # Populate the NFTs
+            api_url_for(
+                rotkehlchen_api_server,
+                'nftsbalanceresource',
+            ),
+            json={'async_query': False, 'ignore_cache': True},
+        )
+        result = assert_proper_response_with_result(response)
+        assert result['entries_found'] == 2
+
+    # Make NFTs queried for only one of the addresses
+    db_queried_addresses = QueriedAddresses(rotki.data.db)
+    db_queried_addresses.add_queried_address_for_module(module='nfts', address=TEST_ACC5)
+
+    response = requests.get(  # Now we should get the NFTs only for TEST_ACC5 address
+        api_url_for(
+            rotkehlchen_api_server,
+            'nftsbalanceresource',
+        ),
+        json={'async_query': False, 'ignore_cache': False},
+    )
+    result = assert_proper_response_with_result(response)
+    assert len(result['entries'][TEST_ACC5]) == 1
+    assert TEST_ACC4 not in result['entries']
```

Comment:
maybe is better to check that 

```suggestion
    result['entries'].keys() == {TEST_ACC5}
```

since it could happen that there were other addresses. Not as it is now but if we change code in the future